source/clear-linux/concepts/debug.rst

```diff
@@ -0,0 +1,89 @@
+.. _debug:
+
+Debug System
+############
+
+|CL-ATTR| introduces a novel approach to system software debugging using
+*clr-debug-info*. On the client side, the |CL| debug system obtains any
+necessary debug information on-the-fly over a network during a debugging
+session. On the server side, the system curates and compresses debug
+information into small pieces for efficient downloading.
+
+For developers, this avoids the interruption during debugging that usually
+happens when debug information is missing. This can be especially useful on
+systems where storage is limited. 
+
+
+.. contents:: :local:
+    :depth: 2
+
+
+Background
+----------
+
+Software that is compiled and packaged for general usage in an operating
+system typically only contains components that are used to execute the
+program, like binaries and libraries. Extra developer data like the the
+actual source code and symbol information are separated and excluded for
+efficiency. 
+
+The debug information helps relate binary code to human readable source code
+lines and variables. Most of the time this auxillary information is not needed
+however without it, debugging a program results in limited visibility. 
+
+
+Usage
+-----
+
+The clr-debug-info system is integrated into |CL| and seamlessly engages once
+installed.
+
+#. Install the *dev-utils* bundle.
+
+   .. code:: bash
+
+      sudo swupd bundle-add dev-utils 
+
+   .. note::
+
+      The *telemetrics* and *performance-tools* bundles also include
+      clr-debug-info.
+
+
+#. Start a debugging session against a program using a debugger, like GDB. 
+   For example, to debug *gnome-control-center*:
+
+   .. code:: bash
+
+      gdb /usr/bin/gnome-control-center
+
+As the program is stepped through and debug information is needed, the
+clr_debug_daemon will obtain it in the background.
+
+
+Implementation
+--------------
+
+The implementation of the |CL| debug system is open source and available on
+GitHub at: https://github.com/clearlinux/clr-debug-info/
+
+.. figure:: figures/debug-diagram.png
+   :scale: 75%
+   :alt: Debug system communication flow
+
+   Figure 1: The communication flow of the |CL| debug system
+
+The |CL| debug system implements a :abbr:`FUSE (filesystem in userspace)`
+filesystem mounted at :file:`/var/cache/debuginfo`. The FUSE filesystem starts
```

Comment:
The FUSE mount points are actually at `/usr/lib/debug` and `/usr/src/debug`, with corresponding cache locations at `/var/cache/debuginfo/lib` and `/var/cache/debuginfo/src`.